groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/api/GroovyIndex.java

```diff
@@ -252,10 +253,18 @@ public Set<IndexedMethod> getConstructors(final String className) {
                     }
                 }
                 int flags = 0;
-                if (parts.length > 2) {
+                if (parts.length > 3) {
```

Comment:
Please document what the `signature` syntax is; from the code it seems that e.g. `type` and `attribute` part may be optional (there are checks for that, but still better to have it documented), the `range` part is (?) mandatory.

Depending on that various checks may be insufficient; e.g. there are (?) 2 optional parts (type, attributes) separted by the same separator. So from notation `name;a` it's not obvious what 'a' is: a type or an attribute ?

But from the Indexer code, it seems that `type` is always present, so maybe the checks in index reading code can be simplified ?

Reply:
Type and range are mondatory. The attribute part is optional. I tried to keep same order as the version before.